vllm/model_executor/models/t5.py

```diff
@@ -0,0 +1,931 @@
+        hidden_states = hidden_states + self.dropout(forwarded_states)
+        return hidden_states
+
+
+class T5Attention(nn.Module):
+
+    def __init__(
+        self,
+        config: T5Config,
+        cache_config: Optional[CacheConfig] = None,
+        quant_config: Optional[QuantizationConfig] = None,
+        has_relative_attention_bias=False,
+        prefix: str = "",
+    ):
+        super().__init__()
+        self.is_decoder = config.is_decoder
+        self.has_relative_attention_bias = has_relative_attention_bias
+        rel_num_bucket = config.relative_attention_num_buckets
+        rel_max_dist = config.relative_attention_max_distance
+        self.relative_attention_num_buckets = rel_num_bucket
+        self.relative_attention_max_distance = rel_max_dist
+        self.d_model = config.d_model
+        self.key_value_proj_dim = config.d_kv
+        self.n_heads = config.num_heads
+        self.dropout = config.dropout_rate
+        self.inner_dim = self.n_heads * self.key_value_proj_dim
+
+        self.qkv_proj = QKVParallelLinear(
+            self.d_model,
+            self.inner_dim // self.n_heads,
+            self.n_heads,
+            bias=False,
+            quant_config=quant_config,
+        )
+        self.out_proj = RowParallelLinear(
+            self.inner_dim,
+            self.d_model,
+            bias=False,
+            quant_config=quant_config,
+        )
+        self.attn = Attention(self.n_heads,
+                              self.inner_dim // self.n_heads,
+                              scale=1,
+                              cache_config=cache_config,
+                              quant_config=quant_config,
+                              prefix=prefix)
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        kv_cache: torch.Tensor,
+        attn_metadata: AttentionMetadata,
+        encoder_hidden_states: Optional[torch.Tensor] = None,
+    ) -> torch.Tensor:
+        """
+        Self-attention (if key_value_states is None) or
+        attention over source sentence (provided by
+        key_value_states).
+        """
+        qkv, _ = self.qkv_proj(hidden_states)
+        q, k, v = qkv.split([2048, 2048, 2048], dim=-1)
+        if encoder_hidden_states is None:
+            attn_output = F.scaled_dot_product_attention(q,
+                                                         k,
+                                                         v,
+                                                         dropout_p=0.0)
+        else:
+            qkv_enc, _ = self.qkv_proj(encoder_hidden_states)
+            _, k, v = qkv.split([2048, 2048, 2048], dim=-1)
+            attn_output = F.scaled_dot_product_attention(q,
+                                                         k,
+                                                         v,
+                                                         dropout_p=0.0)
+        output, _ = self.out_proj(attn_output)
```

Comment:
As @njhill already pointed out, there's no attention bias added here, which is really the main difference of t5. Having no positional encoding, it can't tell tokens positions in the sequence.